packages/clerk-js/src/core/tokenCache.ts

```diff
@@ -75,15 +75,15 @@ const MemoryTokenCache = (prefix = KEY_PREFIX): TokenCache => {
       .then(newToken => {
         const nowSeconds = Math.floor(Date.now() / 1000);
         const expiresAt = newToken.jwt.claims.exp;
-        const expiresIn = expiresAt - nowSeconds;
-
-        if (expiresIn <= 0) {
-          clerkCoreErrorExpiredToken(expiresAt);
-        }
+        const issuedAt = newToken.jwt.claims.iat;
+        const ttl = expiresAt - issuedAt;
+        // if user clock is running slow, expiresAt - nowSeconds will be bigger than the ttl
+        // if user clock is running , expiresAt - nowSeconds will be less than the ttl
+        const expiresIn = expiresAt - nowSeconds <= 0 ? ttl : Math.min(expiresAt - nowSeconds, ttl);
```

Comment:
My instinct is that `expiresIn` should always be `ttl`, otherwise a clock running fast will refresh too frequently.

Also, it might be safer to use the time the promise was initiated instead of when it was resolved. wdyt?